holoviews/core/traversal.py

```diff
@@ -128,3 +128,29 @@ def hierarchical(keys):
                 store1[v2].append(v1)
         hierarchies.append(store2 if hierarchy else {})
     return hierarchies
+
+
+class enable_streams_cache(object):
+    """
```

Comment:
Given the discussion about, I would call it ``disable_cache_lookup`` with constructor:
```
def __init__(self, obj):
```

I don't think it needs an ``enable`` or ``disable`` flag as ``enable`` is the default behaviour.  Unless you can think of situations where you want to be able to disable the context manager with a boolean? If you can think of such a situation (or just think it makes it more flexible) then

```
def __init__(self, obj, disable=True):
```

Would be reasonable. The issue with using either ``enable`` or ``disable`` is it isn't clear if you are enabling/disabling the cache lookup of enabling/disabling the context manager that is enabling/disabling the cache lookup! 

For this reason, maybe something like this would be clearer:

```
def __init__(self, obj, allow_cache_lookup=False):
```

Just some things to think about...